ui/ducks/bridge/actions.ts

```diff
@@ -62,3 +166,359 @@ export const setToChain = (chainId: Hex) => {
     );
   };
 };
+
+export const switchToAndFromInputs =
+  (
+    fromChainId: Hex,
+    fromToken: SwapsTokenObject | SwapsEthToken,
+    toToken: SwapsTokenObject | SwapsEthToken,
+  ) =>
+  async (dispatch: MetaMaskReduxDispatch) => {
+    dispatch(setFromToken(toToken));
+    dispatch(setToChain(fromChainId));
+    dispatch(setToToken(fromToken));
+    dispatch(
+      setFromTokenInputValue({
+        amount: '',
+        decimals: 1,
+      }),
+    );
+  };
+export const getBridgeERC20Allowance = async (
+  contractAddress: string,
+  chainId: Hex,
+): ReturnType<BridgeController['getBridgeERC20Allowance']> => {
+  return await submitRequestToBackground(
+    BridgeBackgroundAction.GET_BRIDGE_ERC20_ALLOWANCE,
+    [contractAddress, chainId],
+  );
+};
+
+export const submitBridgeTransaction = (
+  quoteResponse: QuoteResponse,
+  history: ReturnType<typeof useHistory>,
+) => {
+  return async (
+    dispatch: MetaMaskReduxDispatch,
+    getState: () => MetaMaskReduxState & BridgeAppState,
+  ) => {
+    const state = getState();
+    // Track event TODO
+
+    const calcFeePerGas = () => {
+      let maxFeePerGas: undefined | string;
+      let maxPriorityFeePerGas: undefined | string;
+
+      const networkAndAccountSupports1559 =
+        checkNetworkAndAccountSupports1559(state);
+
+      if (networkAndAccountSupports1559) {
+        const gasFeeEstimates = getGasFeeEstimates(state);
+        maxFeePerGas = decGWEIToHexWEI(
+          gasFeeEstimates?.high?.suggestedMaxFeePerGas,
```

Comment:
Does swaps also use `high`? We only use medium in Portfolio

Reply:
Yes or at least Swaps used to. It was changed in: https://github.com/MetaMask/metamask-extension/pull/27612/files#diff-2493772b1f68622e8d2eae0660b50d2765884331c60418c86e5315389fb2fd8eL1111